### PR TITLE
Basic Documentation for deploying to AWS Lambda

### PR DESCRIPTION
**Pull request by mlocher**
Needs some adaptations for the deployment script in at the end.

### CODE REVIEW

File: _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+
+
+## General
+[AWS Lambda](http://aws.amazon.com/lambda/)  is a compute service that runs your code in response to events and automatically manages the compute resources for you, making it easy to build applications that respond quickly to new information.
+
+## Prerequisites
+
+This deployment method does not create the required configuration on AWS Lambda. Please configure this by hand before you deploy for the first time.
```
Comment (flomotlik) [inline _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md]: Add something like "You can read more about getting setting up your first function in [their Documentation](http://docs.aws.amazon.com/lambda/latest/dg/getting-started.html). Or did you leave that out on purpose?

File: _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## IAM Policies
+
+It is generally a good idea to create a separate [IAM user](http://docs.aws.amazon.com/general/latest/gr/root-vs-iam.html) for Codeship when deploying to AWS. This allows you to explicitly control which resources Codeship can access during your builds. Please take note of the **Access Key ID** and **Secret Access Key** created during the process, as you'll need this when configuring the deployment.
+
+### Lambda
+
+To deploy into Lambda we need to make sure you’ve given us the correct permissions to deploy into it.
```
Comment (flomotlik) [inline _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md]: "Add the following policy to the account on AWS which you've taken the Keys used for deployment from" or something like that to make it clear this is a policy

File: _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+When you go to the Deploy settings of your repository in your Codeship account you now have to add a Script deployment. In the Script deployment, first put the new code you want to deploy into a zip file and then push it to AWS Lambda.
+
+To test this actually works we’ll get the latest function info and invoke the function after the deployment as well. The same command can also be run from your local machine with the invoke script in the repo you’ve forked. Make sure you have permissions set locally that allow you to invoke a function on AWS Lambda.
+
+Following you can see the list of commands to use and how they’ve been added to a script deployment on Codeship.
+
+```bash
+pip install awscli
+zip -r LambdaTest.zip LambdaTest.js
+aws lambda update-function-code --function-name LambdaTest --zip-file fileb://LambdaTest.zip
```
Comment (flomotlik) [inline _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md]: typo. It says fileb for the Lambda file path

**Comment by flomotlik** [comment]
A few comments, but generally looks nice

File: _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md (hunk 3 above)
Comment (mlocher) [inline _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md]: Actually that's what you need to make it work. Took me hours and multiple messages with AWS support to figure that out ;)

Comment (flomotlik) [inline _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md]: ok and I suppose ./LambdaTest.zip doesn't work then. strange but fine

File: _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+---
+
+
+
+## General
+[AWS Lambda](http://aws.amazon.com/lambda/)  is a compute service that runs your code in response to events and automatically manages the compute resources for you, making it easy to build applications that respond quickly to new information.
+
+## Prerequisites
+
+This deployment method does not create the required configuration on AWS Lambda. Please configure this by hand before you deploy for the first time. You can read more about getting setting up your first function on the [AWS Lambda Documentation](http://docs.aws.amazon.com/lambda/latest/dg/welcome.html).
```
Comment (flomotlik) [inline _posts/continuous-deployment/2015-06-16-deployment-to-aws-lambda.md]: getting setting (typo in my sentence copied in). remove getting.

**Comment by flomotlik** [comment]
:shipit:

**Pull request merged**